Default Majority of Configuration

malnick opened the pull request:
Default the configuration we are not passing in via the config.yaml for the DC/OS installer. We should only override things in enterprise config file with EE specific config, everything else can be defaulted for now. TODO - expose collector polling times in config.yaml.

@rji When you get a moment can you review this?

malnick committed:
Specify default configuration

Default the configuration we are not passing in via the config.yaml for
the DC/OS installer. We should only override things in enterprise config
file with EE specific config, everything else can be defaulted for now.

malnick commented on config.go:
```diff
@@ -92,7 +92,8 @@ func (c *Config) setFlags(fs *flag.FlagSet) {
 	fileByte, err := ioutil.ReadFile(c.ConfigPath)
 	if err != nil {
-		return err
+		log.Warnf("%s not found. Using all defaults", c.ConfigPath)
```
Can do this or place logic around `len(c.ConfigPath)` and if it's `> 0` then we try to load config. I'm open to both use cases.

Commit:
misc tweaks around new default config, and tests

rji commented:
@malnick -- I pushed a commit that makes additional changes needed to get this working properly. I also need to write some tests to ensure this doesn't change in the future. I'll take ownership on this PR and ping you when the tests are in place.

Commit:
return errors instead of just logging them

Logging errors and moving on only leads to more errors. If the IP detect
script isn't present, there's no sense in even leaving the process
running. Return the error quickly, and log it as a fatal error (thus
exiting 1 in the process).

Commit:
add tests for getNewConfig()

malnick commented on config.go:
```diff
@@ -174,14 +177,34 @@ func getNewConfig(args []string) (Config, error) {
+		os.Exit(0)
+	}
+
+	if len(c.ConfigPath) > 0 {
```
:+1:

malnick commented:
I had one minor comment about adding the `role` to the version output. Other than that lets :ship: it.

rji commented:
Awesome, thank you sir! I'll add that to the output.

malnick commented:
Oh, one more thing, we're probably only accessing the version command from the CLI - why do we get the user agent from HTTP client?

rji commented:
Yeah, I don't like the user agent output with `-version`, but I do that so that I can test that the linker is replacing the User-Agent string in the build script, and that the final binary doesn't have an "unset" value. I wasn't sure of a nicer way to do this in testing, short of replacing the linker flags with a call out to sed :frowning_face:. If you have any suggestions on how to improve that though, I'd love to hear them!

malnick commented:
Haha, no this is fine.

Commit:
add DC/OS role to -version output
